ui/app/pages/send/send-content/send-asset-row/send-asset-row.component.js

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import SendRowWrapper from '../send-row-wrapper'
+import Identicon from '../../../../components/ui/identicon/identicon.component'
+import TokenBalance from '../../../../components/ui/token-balance'
+import UserPreferencedCurrencyDisplay from '../../../../components/app/user-preferenced-currency-display'
+import {PRIMARY} from '../../../../helpers/constants/common'
+
+export default class SendAssetRow extends Component {
+  static propTypes = {
+    tokens: PropTypes.arrayOf(
+      PropTypes.shape({
+        address: PropTypes.string,
+        decimals: PropTypes.string,
+        symbol: PropTypes.string,
+      })
+    ).isRequired,
+    accounts: PropTypes.object.isRequired,
+    selectedAddress: PropTypes.string.isRequired,
+    selectedTokenAddress: PropTypes.string,
+    setSelectedToken: PropTypes.func.isRequired,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+  }
+
+  state = {
+    isShowingDropdown: false,
+  }
+
+  openDropdown = () => this.setState({ isShowingDropdown: true })
+
+  closeDropdown = () => this.setState({ isShowingDropdown: false })
+
+  selectToken = address => {
+    this.setState({
+      isShowingDropdown: false,
+    }, () => {
+      this.context.metricsEvent({
```

Comment:
@danjm Added a metric event - can you verify that I have used the right category and things?

Note: I am tracking whether or not the user selected a Token or Ether, instead of tracking the exact token address. Erring on the side of privacy.

Reply:
@chikeichan this is correct

In addition, when we first add a custom variable, we need to map it to a specific id in `metamask-extension/ui/app/helpers/utils/metametrics.util.js`

However there are some limitations on how we do that and before we add anymore custom variables I want to reorganize the code and provide clear inline documentation. 

Your code can stay in this PR. But let's not do the additional changes to the metrics util in this PR. I will do them as part of https://github.com/MetaMask/metamask-extension/issues/6453 which I've added to next sprint. I have made a not that references your comment/code.